gammapy/catalog/hawc.py

```diff
@@ -0,0 +1,114 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+HAWC catalogs (https://www.hawc-observatory.org)
+"""
+
+from .core import SourceCatalog, SourceCatalogObject
+from ..utils.scripts import make_path
+from astropy.table import Table
+import numpy as np
+
+class SourceCatalogObject2HWC(SourceCatalogObject):
+    """One source from the HAWC 2FHL catalog.
+
+    Catalog is represented by `~gammapy.catalog.SourceCatalog2HWC`.
+    """
+
+    _source_name_key = 'source_name'
+    _source_index_key = 'catalog_row_index'
+
+    def __str__(self):
+        return self.info()
+
+    def info(self, info='all'):
+        """Summary info string.
+
+        Parameters
+        ----------
+        info : {'all', 'basic', 'position', 'spectrum}
+            Comma separated list of options
+        """
+
+        if info == 'all':
+            info = 'basic,position,spectrum'
+
+        ss = ''
+        ops = info.split(',')
+        if 'basic' in ops:
+            ss += self._info_basic()
+        if 'position' in ops:
+            ss += self._info_position()
+        if 'spectrum' in ops:
+            ss += self._info_spectrum()
+        return ss
+
+    def _info_basic(self):
+        """Print basic info."""
+        d = self.data
+        ss = '\n*** Basic info ***\n\n'
+        ss += 'Catalog row index (zero-based) : {}\n'.format(d['catalog_row_index'])
+        ss += '{:<15s} : {}\n'.format('Source name:', d['source_name'])
+        return ss
+
+    def _info_position(self):
+        """Print position info."""
+        d = self.data
+        ss = '\n*** Position info ***\n\n'
+
+        ss += 'Measurement:\n'
+        ss += '{:20s} : {:.3f}\n'.format('RA', d['ra'])
+        ss += '{:20s} : {:.3f}\n'.format('DEC', d['dec'])
+        ss += '{:20s} : {:.3f}\n'.format('GLON', d['glon'])
+        ss += '{:20s} : {:.3f}\n'.format('GLAT', d['glat'])
+        ss += '{:20s} : {:.3f}\n'.format('Position error', d['pos_err'])
+
+        return ss
+
+    def _info_spectra(self):
+        """Print spectral info."""
+        d = self.data
+        ss = '\n*** Spectral info ***\n\n'
+
+        ss += 'Spectrum 1:\n'
+        ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7 TeV', d['spec0_dnde'].value, \
+            d['spec0_dnde_err'].value, 'cm-2 s-1 TeV-1')
+        ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('Spectral index', d['spec0_index'], d['spec0_index_err'])
+        ss += '{:20s} : {:1}\n\n'.format('Test radius', d['spec0_radius'])
+
+        if(np.isnan(d['spec1_dnde'])):
```

Comment:
Does this work? Shouldn't it be "spec2" here in the if instead of "spec1"?
Can you please, in a comment, share the printout for a source that has only 1 spec, and for a sources that has 2 specs?

Reply:
It works, in the ecsv-file the spectra are called 'spec0' and 'spec1'.
I can change the info-output to
***Spectrum 0***
***Spectrum 1***
but I don't like counting from 0 for non-binary/ digital things.

Reply:
As I said: please past the two examples here so that I can review without having to check locally.

Reply:
This here is a source with only one spectrum:

print(cat[0])

*** Basic info ***

Catalog row index (zero-based) : 0
Source name:    : 2HWC J0534+220

*** Position info ***

Measurement:
RA                   : 83.628 deg
DEC                  : 22.024 deg
GLON                 : 184.547 deg
GLAT                 : -5.783 deg
Position error       : 0.057 deg

*** Spectral info ***

Spectrum 1:
Flux at 7 TeV        : 1.85e-13 +- 2.38e-15 cm-2 s-1 TeV-1
Spectral index       : -2.580 +- 0.010
Test radius          : 0.0 deg

No second spectrum available for this source

and this here a source with two spectra:

*** Basic info ***

Catalog row index (zero-based) : 1
Source name:    : 2HWC J0631+169

*** Position info ***

Measurement:
RA                   : 97.998 deg
DEC                  : 16.997 deg
GLON                 : 195.614 deg
GLAT                 : 3.507 deg
Position error       : 0.114 deg

*** Spectral info ***

Spectrum 1:
Flux at 7 TeV        : 6.71e-15 +- 1.47e-15 cm-2 s-1 TeV-1
Spectral index       : -2.570 +- 0.150
Test radius          : 0.0 deg

Spectrum 2:
Flux at 7 TeV        : 4.87e-14 +- 6.85e-15 cm-2 s-1 TeV-1
Spectral index       : -2.230 +- 0.080
Test radius          : 2.0 deg

Reply:
Perfect!
(I was confused that "1" is the second, you could add a code comment saying that, or leave as-is)